### PR TITLE
Add a tooltip to show the username on the account icon

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] screenshots are included showing significant UI changes

##### Description of change
#796 
Change the tooltip on the account icon in the top right of a page to show the user's username instead. Also, refactor away from currentuser.ts to use the oeq-ts-rest-api.
image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/api/currentuser.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export interface UserData {
   guest: boolean;
   autoLoggedIn: boolean;
   prefsEditable: boolean;
+  username: string;
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/api/currentuser.ts:34]: Aaahhhh, this reminds me.

We need to drop out `currentuser.ts` and use the support in oeq-ts-rest-api. So, this PR seems as good as any.

Could you please remove `currentuser.ts` and move the code to use getCurrentUserDetails (and supporting interfaces) @ https://github.com/openequella/openEQUELLA/blob/develop/oeq-ts-rest-api/src/LegacyContent.ts#L38

File: Source/Plugins/Core/com.equella.core/js/tsrc/mainui/Template.tsx [hunk 2]
```diff
@@ -512,7 +512,7 @@ export const Template = React.memo(function Template({
                 topBarString.notifications
               )}
             </Hidden>
-            <Tooltip title={strings.menu.title}>
+            <Tooltip title={currentUser?.username}>
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/mainui/Template.tsx:515]: Could be good to also have a fallback here - just so we never have an empty "bubble" of a tooltip. Maybe "Username unknown". :thinking:

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Good stuff @SammyIsConfused :+1: 

But it has reminded me of some outstanding refactoring that needs to be done. So let's squeeze that into this PR.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Looks good @SammyIsConfused 

Just two things:

1. Please stick with the convention for importing `@openequella`; and
2. With your renamed `UserDetailsModule.ts` could we please go with `UserModule.ts`. This will align with what's been done in `component/new-search-page` with the [UserModule.ts](https://github.com/openequella/openEQUELLA/blob/component/new-search-page/Source/Plugins/Core/com.equella.core/js/tsrc/modules/UserModule.ts) there. Sure it'll create a bit of a conflict, but it'll highlight it should sit alongside it.

**Review by ChristianMurphy — APPROVED** [review]

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts [hunk 3]
```diff
@@ -237,6 +237,7 @@ export const languageStrings = {
       title: "My Account",
       logout: "Logout",
       prefs: "My preferences",
+      usernameUnknown: "Username unknown",
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts:240]: As we have a language string change log now, do we need to put this new string into the log  when component is merged to develop?

File: Source/Plugins/Core/com.equella.core/js/tsrc/modules/UserDetailsModule.ts [hunk 4]
```diff
@@ -15,41 +15,15 @@
...
-export function getCurrentUser(): Promise<UserData> {
-  return Axios.get<UserData>("api/content/currentuser").then((r) => r.data);
-}
+/**
+ * Gets the current user's info from the server as OEQ.LegacyContent.CurrentUserDetails.
+ */
+export const getCurrentUserDetails = () =>
+  OEQ.LegacyContent.getCurrentUserDetails(API_BASE_URL).then(
+    (result: OEQ.LegacyContent.CurrentUserDetails) => {
+      return result;
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/js/tsrc/modules/UserDetailsModule.ts:27]: Can simplify the callback to `(result: OEQ.LegacyContent.CurrentUserDetails) => result` to save 2 lines :smile:

**Review by PenghaiZhang — APPROVED** [review]
Looks good to me Sam! Thanks.

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts (hunk 3 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts:240]: You raise a good point there @PenghaiZhang 

I was only thinking though to do that with the ResourceCentre ones for now - i.e. the old properly released ones. It feels like all these (`langstrings.ts`) are essentially 'new' for 2020.2. But after that, then we'd want to document these too me thinks.

What do you think?

Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts:240]: That's a good idea! I agree.

File: Source/Plugins/Core/com.equella.core/js/tsrc/modules/UserDetailsModule.ts (hunk 4 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/modules/UserDetailsModule.ts]: Done!

**Review by edalex-ian — APPROVED** [review]
Aw man, looks like the hooks failed on you again. I wonder what that is.

But, aside from that, GTG. :+1: :wink:

**Pull request merged**